src/Controller/EnMarche/MembershipController.php

```diff
@@ -65,12 +66,56 @@ public function registerAction(Request $request, GeoCoder $geoCoder, Authorizati
     /**
      * This action enables a guest user to adhere to the community.
      *
+     * @Route("/adhesion", name="app_membership_register")
      * @Route("/adhesion", name="app_membership_join")
      * @Method("GET|POST")
-     *
-     * @Security("is_granted('ROLE_USER')")
      */
-    public function joinAction(Request $request, EntityManagerInterface $manager): Response
+    public function adhesionAction(Request $request, GeoCoder $geoCoder)
+    {
+        if ($this->isGranted('ROLE_USER')) {
+            return $this->joinAdherent($request);
+        }
+
+        return $this->registerAdherent(
+            $request,
+            $geoCoder
+        );
+    }
+
+    private function registerAdherent(Request $request, GeoCoder $geoCoder): Response
+    {
+        $membership = MembershipRequest::createWithCaptcha(
+            $geoCoder->getCountryCodeFromIp($request->getClientIp()),
+            $request->request->get('g-recaptcha-response')
+        );
+
+        $fromActivation = $request->query->getBoolean('from_activation');
+        $form = $this
+            ->createForm(AdherentRegistrationType::class, $membership)
+            ->add('submit', SubmitType::class, ['label' => 'Adhérer'])
+            ->handleRequest($request)
+        ;
+
+        try {
+            if ($form->isSubmitted() && $form->isValid()) {
+                $user = $this->get(MembershipRequestHandler::class)->handle($membership);
```

Comment:
One e-mail email activation is sent here.
And another is sent the line below because you became adherent.
We should just send one email.
And perhaps the activation e-mail should change if you are a simple User or an Adherent. 
Please check with the team ;-)

Reply:
I guess it would make sense to add a brand new method in the manager for this case.

Reply:
I would prefer that (it's easier to understand and the method name would be clearer for the use case)  than a boolean to prevent email from being sent in the second line.